pkg/lightning/restore/meta_manager.go

```diff
@@ -520,17 +524,81 @@ type storedCfgs struct {
 	RestoreCfg pdutil.ClusterConfig `json:"restore"`
 }
 
-func (m *dbTaskMetaMgr) InitTask(ctx context.Context) error {
+func (m *dbTaskMetaMgr) InitTask(ctx context.Context, source int64) error {
 	exec := &common.SQLWithRetry{
 		DB:     m.session,
 		Logger: log.L(),
 	}
 	// avoid override existing metadata if the meta is already inserted.
-	stmt := fmt.Sprintf(`INSERT IGNORE INTO %s (task_id, status) values (?, ?)`, m.tableName)
-	err := exec.Exec(ctx, "init task meta", stmt, m.taskID, taskMetaStatusInitial.String())
+	stmt := fmt.Sprintf(`INSERT IGNORE INTO %s (task_id, status, source_bytes) values (?, ?, ?)`, m.tableName)
+	err := exec.Exec(ctx, "init task meta", stmt, m.taskID, taskMetaStatusInitial.String(), source)
 	return errors.Trace(err)
 }
 
+func (m *dbTaskMetaMgr) CheckTaskStart(ctx context.Context) (bool, error) {
+	exec := &common.SQLWithRetry{
+		DB:     m.session,
+		Logger: log.L(),
+	}
+	// avoid override existing metadata if the meta is already inserted.
+	hasStarted := false
+	err := exec.Transact(ctx, "check whether this task has started before", func(ctx context.Context, tx *sql.Tx) error {
+		query := fmt.Sprintf("SELECT task_id from %s WHERE task_id = %d", m.tableName, m.taskID)
+		rows, err := tx.QueryContext(ctx, query)
+		if err != nil {
+			return errors.Annotate(err, "fetch task meta failed")
+		}
+		var taskID int64
+		for rows.Next() {
+			if err = rows.Scan(&taskID); err != nil {
+				rows.Close()
+				return errors.Trace(err)
+			}
+			if taskID == m.taskID {
+				hasStarted = true
```

Comment:
I'm not sure if `Transact` will retry the closure. If a retry happens, we actually need to reset `hasStarted` at first.